src/tv2-common/helpers/serverResume.ts

```diff
@@ -0,0 +1,173 @@
+	let currentServerPosition =
+		previousPartEndState?.segmentId === partInstance.segmentId ? previousServerPosition : undefined
+
+	for (const pieceInstance of currentPiecesWithServer) {
+		const pieceDuration =
+			pieceInstance.resolvedDuration ||
+			(setCurrentPieceToNow !== undefined ? setCurrentPieceToNow - pieceInstance.resolvedStart : undefined)
+
+		const content = pieceInstance.piece.content as VTContent | undefined
+		if (pieceInstance.piece.sourceLayerId === SharedSourceLayers.PgmServer && content) {
+			const pieceSeek = content.seek ?? 0
+			const pieceClipEnd = pieceSeek + (pieceDuration ?? 0)
+			const isPlaying =
+				(previousServerPosition?.fileName === content.fileName && previousServerPosition?.isPlaying) || !pieceDuration
+			const serverEndedWithPartInstance = !pieceInstance.resolvedDuration && setCurrentPieceToNow !== undefined
+			currentServerPosition = {
+				fileName: content.fileName,
+				lastEnd: pieceClipEnd,
+				isPlaying,
+				endedWithPartInstance: serverEndedWithPartInstance ? partInstance._id : undefined
+			}
+		} else if (pieceInstance.piece.sourceLayerId === SharedSourceLayers.PgmDVEAdLib) {
+			const serverPlaybackTiming = (pieceInstance.piece.metaData as DVEPieceMetaData | undefined)?.serverPlaybackTiming
+			if (serverPlaybackTiming) {
+				for (const timing of serverPlaybackTiming) {
+					const start =
+						timing.start ??
+						(partInstance.timings?.startedPlayback &&
+							partInstance.timings?.startedPlayback + pieceInstance.resolvedStart)
+					const end =
+						timing.end ??
+						(pieceDuration && partInstance.timings?.startedPlayback
+							? partInstance.timings?.startedPlayback + pieceInstance.resolvedStart + pieceDuration
+							: undefined)
+					const serverEndedWithPartInstance =
+						!pieceInstance.resolvedDuration && setCurrentPieceToNow !== undefined && !timing.end
+					if (currentServerPosition && end && start) {
+						currentServerPosition.lastEnd += end - start
+						currentServerPosition.endedWithPartInstance = serverEndedWithPartInstance ? partInstance._id : undefined
+					}
+				}
+			}
+		}
+	}
+
+	const inTransitionDuration = partInstance.part.inTransition?.previousPartKeepaliveDuration
+	if (
+		currentServerPosition &&
+		!currentPiecesWithServer.length &&
+		inTransitionDuration &&
+		currentServerPosition?.endedWithPartInstance === previousPartEndState?.partInstanceId
+	) {
+		currentServerPosition.lastEnd += inTransitionDuration
+	}
+
+	return currentServerPosition
+}
+
+export function shouldPreservePosition(pieceInstance: IBlueprintResolvedPieceInstance): boolean {
+	return (
+		!!pieceInstance.dynamicallyInserted &&
+		(pieceInstance.piece.sourceLayerId === SharedSourceLayers.PgmServer ||
+			(pieceInstance.piece.sourceLayerId === SharedSourceLayers.PgmDVEAdLib &&
+				!!(pieceInstance.piece.metaData as DVEPieceMetaData | undefined)?.serverPlaybackTiming))
+	)
+}
+
+export function hasTransition(pieceInstance: IBlueprintResolvedPieceInstance): boolean {
```

Comment:
This method doesn't seem to be used anywhere?

Also, it just check whether or not the sourcelayer id is PgmJingle. Unless there is a hidden understanding that being on the PgmJingle source layer means that the piece has a transistion, then there is a mismatch between the name and the implementation